benchmarks/definition/core.py

```diff
@@ -0,0 +1,65 @@
+"""Definition of the Benchmark class."""
+
+import time
+from collections.abc import Callable
+from datetime import datetime
+from typing import Generic, TypeVar
+from uuid import UUID
+
+from attrs import define, field
+from attrs.validators import instance_of, is_callable
+from pandas import DataFrame
+
+from benchmarks.result.metadata_class import ResultMetadata
+from benchmarks.result.result import Result
+
+BenchmarkConfig = TypeVar("BenchmarkConfig")
+BenchmarkFunction = Callable[[BenchmarkConfig], DataFrame]
+
+
+@define
+class Benchmark(Generic[BenchmarkConfig]):
+    """A class to define a benchmark task."""
+
+    name: str = field(validator=instance_of(str))
+    """The name of the benchmark."""
+
+    settings: BenchmarkConfig = field()
+    """The configuration for the benchmark settings."""
+
+    benchmark_callable: BenchmarkFunction = field(validator=is_callable())
+    """The function that executes the benchmark code and returns
+    the results as well as metadata."""
+
+    identifier: UUID = field(validator=instance_of(UUID))
```

Comment:
let's put this as first attribute. I suggest:
* identifier
* name
* _callable
* settings

And then we make settings optional. This gives us the IMHO logical signature `Benchmark(identifier, name, callable, Optional[settings])`

Or what do you guys think?